### PR TITLE
Add SizedBox.square() constructor

### LINKED ISSUE #82613: Introduce SizedBox.square
It would be a handy syntax to

```dart
SizedBox.square(size: 10.0);
```

vs

```dart
SizedBox(width: 10.0, height: 10.0);
```

### PR DESCRIPTION
**Pull request by pedromassango**
Fixes https://github.com/flutter/flutter/issues/82613

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -2248,6 +2248,12 @@ class SizedBox extends SingleChildRenderObjectWidget {
       height = size?.height,
       super(key: key, child: child);
 
+  /// Creates a box whose its [width] and [height] are equals.
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/widgets/basic.dart:2251]:
```suggestion
  /// Creates a box whose [width] and [height] are equal.
```

File: packages/flutter/test/widgets/basic_test.dart [hunk 2]
```diff
@@ -622,6 +622,20 @@ void main() {
     expect(logs, <String>['exit3']);
     logs.clear();
   });
+
+  testWidgets('SizedBox.square tests', (WidgetTester tester) async {
```
Comment (dkwingsmt) [inline packages/flutter/test/widgets/basic_test.dart:626]: There's a `sized_box_test.dart` where this test should be put in.

**Review by dkwingsmt — CHANGES_REQUESTED** [review]
Other than the comments, I don't have the say on whether we should add this. @goderbauer Do you think we should add this?

**Review by HansMuller — APPROVED** [review]
Although this only seems marginally useful, we already have `Size.square()`, so OK.

**Pull request merged**